frontend/src/components/pages/HomePage/LogRecordsTable.tsx

```diff
@@ -8,66 +8,243 @@ import {
+  const { authenticatedUser, setAuthenticatedUser } = useContext(AuthContext);
+
+  const [showAlert, setShowAlert] = useState(false);
+
+  // Menu states
+  const [deleteOpenMap, setDeleteOpenMap] = useState<{
+    [key: number]: boolean;
+  }>({});
+  const [editOpenMap, setEditOpenMap] = useState<{ [key: number]: boolean }>(
+    {},
+  );
+
+  // Dropdown option states
+  const [employeeOptions, setEmployeeOptions] = useState<UserLabel[]>([]);
+  const [residentOptions, setResidentOptions] = useState<UserLabel[]>([]);
+
+  // Handle delete confirmation toggle
+  const handleDeleteToggle = (logId: number) => {
+    setDeleteOpenMap((prevDeleteOpenMap) => ({
+      ...prevDeleteOpenMap,
+      [logId]: !prevDeleteOpenMap[logId],
+    }));
+  };
+
+  // Handle edit form toggle
+  const handleEditToggle = (logId: number) => {
+    setEditOpenMap((prevEditOpenMap) => ({
+      ...prevEditOpenMap,
+      [logId]: !prevEditOpenMap[logId],
+    }));
+  };
+
+  // fetch resident + employee data for log creation
+  const getLogEntryOptions = async () => {
+    const residentsData = await ResidentAPIClient.getResidents({
+      returnAll: true,
+    });
+
+    if (residentsData && residentsData.residents.length !== 0) {
+      // TODO: Remove the type assertions here
+      const residentLabels: UserLabel[] = residentsData.residents.map((r) => ({
+        id: r.id!,
+        label: r.residentId!,
+        value: r.id!,
+      }));
+      setResidentOptions(residentLabels);
+    }
+
+    const usersData = await UserAPIClient.getUsers({ returnAll: true });
+    if (usersData && usersData.users.length !== 0) {
+      const userLabels: UserLabel[] = usersData.users
+        .filter((user) => user.userStatus === "Active")
+        .map((user) => ({
+          id: user.id,
+          label: user.firstName,
+          value: user.id,
+        }));
+      setEmployeeOptions(userLabels);
+    }
+  };
+
+  useEffect(() => {
+    if (showAlert) {
+      setTimeout(() => {
+        setShowAlert(false);
+      }, 3000);
+    }
+  }, [showAlert]);
+
+  useEffect(() => {
+    getLogEntryOptions();
+  }, []);
+
   return (
-    <Box>
-      <TableContainer
-        marginTop="12px"
-        height="70vh"
-        overflowY="scroll"
-        ref={tableRef}
+    <>
+      <Box>
+        <TableContainer
+          marginTop="12px"
+          height="70vh"
+          overflowY="scroll"
+          ref={tableRef}
+        >
+          <Table variant="showTable" minHeight="400px" verticalAlign="middle">
+            <Thead>
+              <Tr>
+                <Th>Date</Th>
+                <Th>Time</Th>
+                <Th>Resident</Th>
+                <Th>Note</Th>
+                <Th>Employee</Th>
+                <Th>Attn To</Th>
+                <Th> </Th>
+              </Tr>
+            </Thead>
+
+            <Tbody>
+              {logRecords.map((record) => {
+                // TODO: Investigate alternative methods for date storage + creation
```

Comment:
nit: While we're here, lets yeet this comment into oblivion (I added this initially because I thought there would be a performance degradation due to spam creating date objs but I think I just overthought this)